### DISCUSSION TITLE
Toggle to turn on new tab focus instead of search box at start up

### DISCUSSION THREAD

**Discussion by loustact** [zen-browser/desktop#1169] [Ideas]
I use a home menu that has its own search box. Every time I use Firefox, it would always focus on the custom startpage's search box. But now when I tried it with Zen, it would always focus on the leftmost end of the URL bar, which is a bit annoying. It would be nice if I can have an option where it instead focuses on the custom startpage.

**Comment by xburrito** [comment]
Having the exact same problem. And honestly feels more like a bug. Even if I use a generic homepage like google, the moment I launch in, the text cursor auto focuses on the beginning of the url bar in front of https. Like... why? lol

**Comment by Matt-FTW** [comment]
The issue is related to [this line](https://github.com/zen-browser/desktop/blob/996c2929161613d6959b4ff2a667515a1969db63/src/browser/base/content/ZenStartup.mjs#L11) on the startup file which always calls the function that focuses on the url bar.

**Comment by Matt-FTW** [comment]
PR created: #1742

**Reply by Soham52** [reply]
Since Mauro has closed the PR, do you think that you can add an toggle by yourself? maybe then he'd accept it :(

**Comment by TheStutterNerd** [comment]
Ngl I always thought this were some bug. Not a "feature" because it doesn't focus the bar and select the text allowing you to overwrite it like other browsers. It focuses the bar at the left most position so when you start typing, the text gets entered before the url which is silly. Not sure how that's not classed as a bug. It should either focus and select the text or nothing at all.

**Comment by sekiryl** [comment]
If anyone finds a temporary solution to this please let me know. I really want to switch to zen browser but stuck because of this as I am a full keyboard user.

**Comment by TheStutterNerd** [comment]
The workaround I've found is to download the new tab override extention. It allows you to force focus on the page.

**Reply by sekiryl** [reply]
I am already using it but it only works when opening new tabs. When launching the browser it still focuses on address bar. The very temporary workaround I am using for it is using a script which presses `Alt + Home` using ydotool after launching the browser.

**Comment by bon-sai-tree** [comment]
Created a new PR which adds an toggle to control the startup focus: https://github.com/zen-browser/desktop/pull/2920

**Comment by AggressiveHayBale** [comment]
What is the status of this PR? The autofocus is an extremely annoying bug especially with vim keybinding enabled (vimium) and tilling WM when I stay away from the mouse. The focus not even remove a current URL but modify the existing one as such is just a useless when you use any homepage with url.
